test/performance/scale_from_zero_test.go

```diff
@@ -71,10 +70,7 @@ func runScaleFromZero(idx int, t *testing.T, clients *test.Clients, ro *test.Res
 		clients.KubeClient,
 		t.Logf,
 		domain,
-		pkgTest.Retrying(pkgTest.MatchesAllOf(pkgTest.IsStatusOK, pkgTest.MatchesBody(helloWorldExpectedOutput)),
-			http.StatusNotFound,            /* revision not created */
-			http.StatusInternalServerError, /*revision not yet ready*/
-		),
+		pkgTest.MatchesAllOf(pkgTest.IsStatusOK, pkgTest.MatchesBody(helloWorldExpectedOutput)),
```

Comment:
🤔  My understanding was that `WaitFor` retried until the state was met, but perhaps that is only with `pkgTest.Retrying` and the status codes to retry?  Does `MatchesAllOf` just error / succeed without any retrying?  What roll does the timeout play in all of this?

Reply:
It will poll every Nms on timeout errors until timeout is reached.
then the response is passed to the checker wherein retrying would retry on given status codes (404/503). Now with that one we match the service shape and served value and fail on first non 200 response.